### PR TITLE
Add Support For Newer Python Versions

### PR DESCRIPTION
**Pull request by MichaelYochpaz**
## Related Issues
fixes: [CIAC-9602](https://jira-dc.paloaltonetworks.com/browse/CIAC-9602)
fixes: [CIAC-9614](https://jira-dc.paloaltonetworks.com/browse/CIAC-9614)

## Description
This PR updates the `demisto-py` package to support Python versions 3.11 and 3.12.

[SDK nightly build using this branch](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/811446)
[Content nightly build using this branch](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/811445)

### CODE REVIEW

File: .flake8
Comment (MichaelYochpaz) [inline .flake8:1]: Removed as we don't use flake8's linting anywhere (probably a leftover from when we did in the past)

File: docs/DefaultApi.md
Comment (MichaelYochpaz) [inline docs/DefaultApi.md:1]: This file was automatically updated with these changes by `gen-code.sh`

File: pyproject.toml [hunk 1]
```diff
@@ -16,25 +16,25 @@ classifiers=[
...
-certifi = ">=2017.4.17"
-six = ">=1.10"
-python-dateutil = ">=2.5.3"
-urllib3 = ">=1.26.7"
-tzlocal = ">=4.0.0,<5.0.0"
+python = ">=3.8,<=3.12"
+certifi = "^2023.11.17"
+six = "^1.16"
+python-dateutil = "^2.7.2"
+urllib3 = "<2.1.0"
```
Comment (MichaelYochpaz) [inline pyproject.toml:29]: We can't update to versions 2.1.0+ until [CIAC-9648](https://jira-dc.paloaltonetworks.com/browse/CIAC-9648) is resolved.

File: tests/mocks_test.py [hunk 2]
```diff
@@ -36,240 +82,211 @@ def get_login_mocker(mocker, num_of_http_requests_mock: int=0):
...
+
+    # Create an instance of the API class
+    api_instance = demisto_client.configure(base_url=host, api_key=api_key)
+    request_mock = mocker.patch.object(api_instance.api_client.rest_client, 'request', return_value=mock_response)
+
+    create_incident_request = demisto_client.demisto_api.CreateIncidentRequest()
+    create_incident_request.name = 'Test Incident'
+    create_incident_request.type = 'Unclassified'
+    create_incident_request.owner = 'Admin'
+    create_incident_request.occurred = dt.datetime.now()
```
Comment (shmuel44) [inline tests/mocks_test.py:154]: Maybe we need to add
`@freeze_time("2024-01-15 17:00:00 UTC")`

**Review by shmuel44 — COMMENTED** [review]
Good Job!

File: docs/DefaultApi.md [hunk 3]
```diff
@@ -1172,14 +1172,14 @@ Exports an incidents batch to CSV file (returns file ID)
...
+import demisto_client
 import demisto_client.demisto_api
 from demisto_client.demisto_api.rest import ApiException
 from pprint import pprint
 
 api_instance = demisto_client.configure(base_url="https://YOUR_DEMISTO_SERVER", api_key="YOUR_API_KEY")
-inc_filter = demisto_client.demisto_api.IncidentFilter()
-inc_filter.query = 'status: "Active"'
-update_data_batch = demisto_client.demisto_api.UpdateDataBatch(filter=inc_filter, all=True, columns=['id', 'name', 'status'])
+update_data_batch = demisto_client.demisto_api.UpdateDataBatch() # UpdateDataBatch |  (optional)
```
Comment (amshamah419) [inline docs/DefaultApi.md:1182]: Let's revert this change. I think someone didn't update the gen_code.sh script

**Review by amshamah419 — APPROVED** [review]
Good work!

**Review by bziser — APPROVED** [review]
👏
